overlord/ifacestate/ifacemgr.go

```diff
@@ -62,6 +62,7 @@ func Manager(s *state.State, extra []interfaces.Interface) (*InterfaceManager, e
 	runner.AddHandler("setup-profiles", m.doSetupProfiles, m.doRemoveProfiles)
 	runner.AddHandler("remove-profiles", m.doRemoveProfiles, m.doSetupProfiles)
 	runner.AddHandler("discard-conns", m.doDiscardConns, m.undoDiscardConns)
+	runner.AddHandler("discard-namespace", m.doDiscardNamespace, nil)
```

Comment:
Implementation note: should discard the namespace _after_ the snap file and the respective mount units are removed, as this is the point of no return.